met-web/src/components/survey/create/CloneOptions.tsx

```diff
@@ -0,0 +1,232 @@
+                form_json: selectedEngagement ? selectedEngagement.surveys[0].form_json : selectedSurvey.form_json,
+            });
+
+            dispatch(
+                openNotification({
+                    severity: 'success',
+                    text: 'Survey created, please proceed to building it',
+                }),
+            );
+            navigate(`/surveys/${createdSurvey.id}/build`);
+        } catch (error) {
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Error occurred while attempting to save survey',
+                }),
+            );
+            setIsSaving(false);
+        }
+    };
+
+    return (
+        <Grid container direction="row" alignItems="flex-start" justifyContent="flex-start" item xs={12} spacing={2}>
+            <Grid item xs={6}>
+                <MetLabel sx={{ marginBottom: '2px', display: 'flex' }}>
+                    Select Engagement <Typography sx={{ ml: 1, color: '#ACA9A9' }}>(optional)</Typography>
+                </MetLabel>
+
+                <Autocomplete
+                    id="engagement-selector"
+                    options={availableEngagements || []}
+                    renderInput={(params) => (
+                        <TextField
+                            {...params}
+                            label=" "
+                            InputLabelProps={{
+                                shrink: false,
+                            }}
+                            fullWidth
+                            onChange={(e) => {
+                                handleFetchEngagements(1, 10, 'asc', e.target.value);
+                            }}
+                        />
+                    )}
+                    getOptionLabel={(engagement: Engagement) => engagement.name}
+                    onChange={(_e: React.SyntheticEvent<Element, Event>, engagement: Engagement | null) => {
+                        setSelectedEngagement(engagement);
+                        if (engagement !== null) setSelectedSurvey(engagement.surveys[0]);
+                    }}
+                    disabled={loadingEngagements}
+                />
+            </Grid>
+            <Grid item xs={12}>
+                <MetLabel sx={{ marginBottom: '2px' }}>Select Survey</MetLabel>
+                <Autocomplete
+                    id="survey-selector"
+                    options={availableSurveys || []}
+                    renderInput={(params) => (
+                        <TextField
+                            {...params}
+                            label=" "
+                            value={selectedEngagement && selectedSurvey != null ? selectedSurvey.name : ''}
+                            InputLabelProps={{
+                                shrink: false,
+                            }}
+                            onChange={(e) => {
+                                handleFetchSurveys(1, 10, 'asc', e.target.value);
```

Comment:
best to  debounce or throttle this since on each keystroke an API call will happen